### ISSUE TITLE
Voice message button in the app

### ISSUE THREAD

**Issue by the-cave-dweller** [Nheko-Reborn/nheko#1157]
### The Problem

I've currently been bridging my matrix with whatsapp and other platforms, and noticed that the only way I was able to send my contacts any kind of recorded message was trough the attachment menu.

### The Solution

I noticed sending mp3 files via the attachment menu worked fine, as this format actually worked great on all platforms, exactly like the voice recordings on the main element client, so I thought that maybe suggesting a button that sends some mainstream audio formats recording directly would be great for chatting in a more confy way, as saying something large or complicated sometimes tends to be quite time consuming on text.

### Happens in the latest version

- [X] Yes, this feature is still missing.

**Comment by pizdjuk** [comment]
This is the feature that I miss more then video-calls on Pinephone.

**Comment by BeppeInfo** [comment]
I'd like to take a run at this. Starting from the bottom: voice messages need `org.matrix.msc3245.voice` and `org.matrix.msc1767.audio` ([MSC3245](https://github.com/matrix-org/matrix-spec-proposals/pull/3245) / [MSC3246](https://github.com/matrix-org/matrix-spec-proposals/pull/3246)) on the `m.audio` content, and mtxclient doesn't model either, so nheko currently can't send a voice message or tell one apart from an attached audio file on receive.

I've opened https://github.com/Nheko-Reborn/mtxclient/pull/115 against mtxclient to add them. Once that lands, the nheko side looks fairly contained — the upload path already takes an arbitrary `QIODevice`, so a `QMediaRecorder` writing into a `QBuffer` can feed `InputBar::startUpload()` directly with no temp file, and `finalizeUpload()` already routes `audio/*` to `InputBar::audio()`.

Two things I'd like a maintainer's read on before I write much of the nheko side:

1. **Recording backend.** NeoChat uses `QMediaCaptureSession` + `QAudioInput` + `QMediaRecorder` with Ogg/Opus, and Qt6 Multimedia is already a hard dependency here. But available codecs depend on which backend Qt was built against, which varies across the platforms nheko ships on. Is Qt Multimedia with an `isSupported()` check and the button hidden when Opus isn't available acceptable, or would you rather this went through GStreamer (`opusenc` is already used in `WebRTCSession`) despite `-DVOIP=OFF` being a supported build?

2. **Waveform.** I'd like to skip generating one in the first iteration, since nheko has no waveform playback UI and the marker alone is what makes other clients render it as a voice message. NeoChat also ships an empty waveform. It can be added later as a `QAudioDecoder` post-pass without touching the record path. Any objection?

Happy to adjust either way — just don't want to build the wrong thing.

**Comment by spantaleev** [comment]
@BeppeInfo, it seems like you're using Claude to add additional functionality to Nheko.

At [Komai](https://github.com/etkecc/komai) we've already done this work a few months ago and voice messages (along with [speech-to-text dictation](https://github.com/etkecc/komai/blob/main/docs/user-guide/features/voice-transcription.md)) is already a supported feature.

Instead of reinventing the wheel via pull requests that are likely to go nowhere, consider giving Komai a try! Of course, feel free to stay on this path as well, if this makes you happier!

**Comment by BeppeInfo** [comment]
@spantaleev There are already other clients with voice message support like Element and Neochat

I just want is specifically for Nheko since it's the one that works best for me

Since Komai is based on Nheko, have you guys tried upstreaming this feature?

**Comment by spantaleev** [comment]
@BeppeInfo: my point was that Komai is basically "A fork of Nheko with a bunch of stuff like this", so you may consider it working well enough for you as well. These other apps are completely different, so I was not suggesting "move to a completely different app that has the feature already", but rather "try this almost-Nheko-like thing that has the feature - you may like it".

Of course, Komai is also quite different from Nheko (possibly in directions you don't like), so your point probably stands.

**Comment by BeppeInfo** [comment]
@spantaleev I have started trying Komai. Pretty neat project, i might switch to it on my desktop

Nheko still seems to be more lightweight, so I'll keep it on my Raspberry Pi workstation
